components/NVS/src/Storage.cpp

```diff
@@ -100,6 +100,18 @@ esp_err_t NVS::get_bool(const char* key, bool& value)
     return ESP_OK;
 }
 
+esp_err_t NVS::setString(std::string key, std::string data) 
+{
+    return nvs_set_str(m_handle, key.c_str(), data.c_str());
+}
+
+esp_err_t NVS::getString(std::string key,  char* buffer, size_t* length)
+{
+
+    return  nvs_get_str(m_handle, key.c_str(), buffer, length);
```

Comment:
```suggestion
esp_err_t NVS::getString(const char* key,  std::string* data)
{
    esp_err_t err;
    std::size_t required_size;
    if ((err = nvs_get_str(m_handle, key.c_str(), nullptr, &required_size)) != ESP_OK) {
        return err;
    }
    data->clear();
    data->reserve(required_size);
    data->resize(required_size, '\0');
    return  nvs_get_str(m_handle, key, data->data(), &required_length);
```

Reply:
For type safety, convert directly to a `std::string` and to maintain consistency. Can be done later if you want. I'm ok merging this PR as is.